x/ibc/02-client/exported/codec.go

```diff
@@ -0,0 +1,132 @@
+package exported
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	connectionexported "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/exported"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+)
+
+// Codec defines the interface required to serialize custom x/ibc types.
+type Codec interface {
```

Comment:
I'm not sure how I feel about continuing to add `Codec` interfaces for new modules that were never in a previous amino release of the SDK. If we're going to go with Any, why not just do the marshaling right in the keeper. What do you think @alexanderbez ?

Reply:
Yes, I absolutely agree.